modules/ROOT/pages/websocket/websocket-connector.adoc

```diff
@@ -0,0 +1,145 @@
++
+In this scenario, the Mule app initiates communication and asks an external system to establish a WebSocket.
+
+[[socket_id]]
+=== Socket ID
+
+All WebSockets, both inbound and outbound, have a unique socket ID that identifies them. The app must reference these IDs to operate on individual WebSockets.
+
+[[groups]]
+=== Groups
+
+WebSocket Connector introduces the concept of groups, which are string identifiers that individual WebSockets can optionally subscribe to. An app can use groups to broadcast one single message to a group of different WebSockets. Groups facilitate the process of identifying which WebSockets should participate in each broadcast.
+
+Subscriptions only exist in the context of each instance of a Mule WebSocket Connector configuration. Therefore, when you establish a WebSocket between a Mule app and a remote system, subscribing the WebSocket on Mule's end of the connection to a group doesn't affect the WebSockets on the remote system's end.
+
+[[triggers]]
+== Triggers
+
+Triggers initiate the execution of a flow upon receiving a message or event notification. You can specify the following triggers for WebSocket-related events:
+
+* `inbound-listener` exposes an endpoint that external systems can connect to and initiates the containing flow when a message is received.
+* `on-inbound-connection` initiates a flow when a new connection to the Inbound Listener is established.
+* `outbound-listener` exposes an endpoint that initiates a flow when an outbound WebSocket receives a message from an external system.
+* `on_connection-closed` initiates the containing flow when an inbound or outbound WebSocket is closed.
+
+[[operations]]
+== Operations
+
+WebSocket Connector support the following operations:
+
+* `open-outbound-socket` connects to an external system and creates an outbound webSocket.
+* `send` sends a message through a WebSocket.
+* `subscribe-groups` subscribes a WebSocket to one or more groups.
++
+The app can invoke this operation on the same web socket multiple times. each invocation being additive over the previous ones. Repeated groups are ignored.
++
+* `unsubscribe-groups` unsubscribes a WebSocket from one or more groups.
++
+You can invoke this operation multiple times on the same socket, with each invocation being additive over the previous ones. Repeated groups are ignored.
++
+* `broadcast` sends a message to multiple WebSockets classified by path, type, and group.
+* `close-socket` closes a WebSocket's connection.
+
+[[security]]
+== Security
+
+WebSocket Connector supports securing the client and server endpoints through TLS, using the following URI schemes:
+
+* WS identifies a WebSocket server and resource name without using encryption (for example: `ws://localhost:80`).
+* WSS identifies a WebSocket server and resource name, secure through TLS (for example: `wss://localhost:433`).
+
+
+== See Also
+
+* xref:websocket/websocket-connector-key-concepts.adoc[WebSocket Connector Key Concepts]
```

Comment:
The Key Concepts was one of Diane's topics that we didn't pull in (I am pretty sure). Delete this link. Maybe link to the reference.